portal-impl/src/com/liferay/portal/service/impl/LayoutLocalServiceHelper.java

```diff
@@ -104,16 +121,25 @@ public String getFriendlyURL(
 
 		for (int i = 1;; i++) {
 			try {
+				_layoutFriendlyURLValidator = _serviceTracker.getService();
+
 				validateFriendlyURL(
 					groupId, privateLayout, layoutId, friendlyURL, languageId);
+				_layoutFriendlyURLValidator.validateFriendlyURLEntry(
```

Comment:
For example, if we disable the LayoutFriendlyURLValidator it could be null and a NPE could be thrown here